server/src/common/dto/xp-systems/xp-systems.dto.ts

```diff
@@ -0,0 +1,171 @@
+﻿import { ApiProperty, PickType } from '@nestjs/swagger';
+import { NestedDto } from '@lib/dto.lib';
+import { IsNumber } from 'class-validator';
+import { Prisma } from '@prisma/client';
+
+// Could implement RankXpParams and CosXpParams in xp-systems.interface.ts but class-validator requires
+// we give it this wacky structure.
+
+// TODO: bunch of this can be readonly?
+class CosXpTierScale {
+    [key: string]: Prisma.JsonValue;
```

Comment:
I'm really confused by this, though I confess I don't think I've seen index signatures in classes before. I would assume it means "any string-based key must be a `Prisma.JsonValue`" but clearly we have values of other types in here. Can you explain to me what it's needed, either on here or Discord?